processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -296,8 +296,18 @@ else if ( !method.isUpdateMethod() ) {
                             existingVariableNames
             );
 
-            if (factoryMethod != null && method instanceof ForgedMethod ) {
-                ( (ForgedMethod) method ).addThrownTypes( factoryMethod.getThrownTypes() );
+            if ( method instanceof ForgedMethod ) {
+                ForgedMethod forgedMethod = (ForgedMethod) method;
+                if ( factoryMethod != null ) {
```

Comment:
and what about factory methods?

Reply:
What about them? Before the if check was `if ( factoryMethod != null && method instanceOf ForgedMethod )`. I moved the `if ( factoryMethod != null )` inside.

Reply:
I meant: what if a factory method throws an exception while creating an object.. but I guess that's covered elsewhere ...

Reply:
I didn't go down this rabbit hole :). I guess it is covered in some other test case